### ISSUE TITLE
Social Icons: Add editing colors to the Block Toolbar

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#56369]
## What problem does this address?
It's very common for users to want to edit the color of their social icons, but this action is quite hard to find - you have to select the parent block, then move to the styles tab, then change the color.

<img width="1257" alt="Screenshot 2023-11-21 at 10 30 57" src="https://github.com/WordPress/gutenberg/assets/275961/bba714e3-0e76-491f-b9dc-41d09cf43b64">

## What is your proposed solution?
We could move the “colors” settings to the Block Toolbar as they are likely a principle action.

**Comment by hanneslsm** [comment]
I think it should be the other way around. We should rather put the font size to the sidebar. I always wondered why it's in the toolbar, since we have (nearly) everywhere the styles in the sidebar and only "quick actions" in the toolbar. Then, we'd had consistency again and all styling aspects would be in the sidebar in the "styles" tab.

**Comment by colinduwe** [comment]
To clarify, we have two blocks, the parent block core/social-links which has support for icon color and background color. While not perfectly intuitive how one selects this parent block you can then set the style of all child blocks, resulting in a consistent appearance.

<img width="366" alt="Screenshot 2023-12-18 at 3 34 33 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/90ccd867-3cbe-4d39-a55c-dbe93073b353">

We also have the child block, core/social-link with lots of variations for various social media platforms. There is a default black icon color and grey background color (see phone and mail) and many variations define specific colors appropriate to their brand (Twitter, Facebook, Instagram).
<img width="375" alt="Screenshot 2023-12-18 at 3 41 48 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/ae6b2e2b-9aca-404a-b98f-7b30c1a76714">
But the individual child block cannot be assigned a color or background color
<img width="285" alt="Screenshot 2023-12-18 at 3 43 40 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/598ab322-5eca-4106-ba6a-b40463420225">

The enhancement to add the color and background color controls to the core/social-link block.

I think this would be helpful enhancement. For example, in in the image above I would imagine a user would prefer to change the default background color of the phone and mail links so they are distinct from the page background color.

I would argue against removing the color and background color controls from the core/social-links block so we can continue to style them in a central place. I think the styles set on the core/social-link block should take precedence over the style set on the parent.

I'm happy to make a PR for this if any committers think we're on the right track.

**Comment by Mamaduka** [comment]
I agree with @hanneslsm. The color controls are placed in the sidebar by default; moving them into the block toolbar will introduce inconsistency. The Size control changes are tracked in #65647.

I'm going to close this as "won't fix" / "maybe later".

@colinduwe, that seems to be a different proposal. Do you want to create a separate issue if it doesn't exist? We can move the discussion there.
